src/Lecture3.hs

```diff
@@ -79,7 +88,13 @@ Tuesday
   would work for **any** enumeration type in Haskell (e.g. 'Bool',
   'Ordering') and not just 'Weekday'?
 -}
-next = error "TODO"
+next :: forall a. (Enum a, Bounded a) => a -> a
+next = toEnum . (`mod` size) . (+ 1) . fromEnum
+  where size = 1 + fromEnum (maxBound :: a)
```

Comment:
That looks like a correct solution 👍🏻 
But it also looks like overkill for such a simple task.

You can use the `succ` method of the `Enum` typeclass that returns the next value. The only problem with `succ` is that it throws exceptions on the last value of an enumeration. So you need to handle this case separately.

Reply:
very nice, thank you!